OpenRA.Mods.Common/Widgets/Logic/Lobby/LatencyTooltipLogic.cs

```diff
@@ -0,0 +1,48 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Graphics;
+using OpenRA.Network;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public class LatencyTooltipLogic : ChromeLogic
+	{
+		[ObjectCreator.UseCtor]
+		public LatencyTooltipLogic(Widget widget, TooltipContainerWidget tooltipContainer, OrderManager orderManager, int clientIndex)
+		{
+			var latencyPrefix = widget.Get<LabelWidget>("LATENCY_PREFIX");
+			var latencyPrefixFont = Game.Renderer.Fonts[latencyPrefix.Font];
+			var latency = widget.Get<LabelWidget>("LATENCY");
+			var latencyFont = Game.Renderer.Fonts[latency.Font];
+			var rightMargin = widget.Bounds.Width;
+
+			latency.Bounds.X = latencyPrefix.Bounds.X + latencyPrefixFont.Measure(latencyPrefix.Text + " ").X;
+
+			widget.IsVisible = () => orderManager.LobbyInfo.ClientWithIndex(clientIndex) != null;
+			tooltipContainer.BeforeRender = () =>
+			{
+				if (widget.IsVisible())
+					widget.Bounds.Width = latency.Bounds.X + latencyFont.Measure(latency.GetText()).X + rightMargin;
+			};
+
+			var client = orderManager.LobbyInfo.ClientWithIndex(clientIndex);
+			var ping = orderManager.LobbyInfo.PingFromClient(client);
+			if (latency != null)
```

Comment:
This check doesn't make sense to me. If `latency` is `null`, we'll have crashed a few lines before.

Reply:
Nvm, this seems to be changed in later commits.